mlflow/store/db_migrations/versions/cfd24bdc0731_update_run_status_constraint_with_killed.py

```diff
@@ -5,70 +5,51 @@
     RunStatus.to_string(RunStatus.FINISHED),
     RunStatus.to_string(RunStatus.RUNNING),
     RunStatus.to_string(RunStatus.KILLED)
 ]
 
-previous_run_status = [
-    RunStatus.to_string(RunStatus.SCHEDULED),
-    RunStatus.to_string(RunStatus.FAILED),
-    RunStatus.to_string(RunStatus.FINISHED),
-    RunStatus.to_string(RunStatus.RUNNING)
+# Certain SQL backends (e.g., SQLite) do not preserve CHECK constraints during migrations.
+# For these backends, CHECK constraints must be specified as table arguments. Here, we define
+# the collection of CHECK constraints that should be preserved when performing the migration.
+# The "status" constraint is excluded from this set because it is explicitly modified
+# within the migration's `upgrade()` routine.
+check_constraint_table_args = [
+    CheckConstraint(SqlRun.source_type.in_(SourceTypes), name='source_type'),
+    CheckConstraint(SqlRun.lifecycle_stage.in_(LifecycleStage.view_type_to_stages(ViewType.ALL)),
+                    name='runs_lifecycle_stage'),
 ]
 
 
-def _has_check_constraints():
-    inspector = Inspector(context.get_bind().engine)
-    constraints = [constraint['name']
-                   for constraint in inspector.get_check_constraints("runs")]
-    return "status" in constraints
-
-
-def get_check_constraints():
-    """
-    Get all check constraint except status
-    """
-    inspector = Inspector(context.get_bind().engine)
-    constraints = [CheckConstraint(constraint['sqltext'], name=constraint['name'])
-                   for constraint in inspector.get_check_constraints("runs")
-                   if constraint['name'] != 'status']
-    return constraints
-
-
 def upgrade():
-    if _has_check_constraints():
-        with op.batch_alter_table("runs", table_args=get_check_constraints()) as batch_op:
-            # Update the "status" constraint via the SqlAlchemy `Enum` type. Specify
-            # `native_enum=False` to create a check constraint rather than a
-            # database-backend-dependent enum (see https://docs.sqlalchemy.org/en/13/core/
-            # type_basics.html#sqlalchemy.types.Enum.params.native_enum)
-            batch_op.alter_column("status",
-                                  type_=Enum(*new_run_status, create_constraint=True,
-                                             constraint_name="status", native_enum=False))
+    with op.batch_alter_table("runs", table_args=check_constraint_table_args) as batch_op:
+        # Transform the "status" column to an `Enum` and define a new check constraint. Specify
+        # `native_enum=False` to create a check constraint rather than a
+        # database-backend-dependent enum (see https://docs.sqlalchemy.org/en/13/core/
+        # type_basics.html#sqlalchemy.types.Enum.params.native_enum)
+        batch_op.alter_column("status",
+                              type_=Enum(*new_run_statuses, create_constraint=True,
+                                         constraint_name="status", native_enum=False))
```

Comment:
Somehow the constraint name here doesn't seem to be honored - as a fresh user, on MySQL I get a constraint named "runs_chk_1":

```
  CONSTRAINT `runs_chk_1` CHECK ((`status` in (_utf8mb4'SCHEDULED',_utf8mb4'FAILED',_utf8mb4'FINISHED',_utf8mb4'RUNNING',_utf8mb4'KILLED'))),
```

& @AveshCSingh gets a different constraint name on postgres. As far as I can tell, this is what seems to be happening:
* For a MySQL user who previously upgraded to 1.5.0, we (in the old migration logic) created a new constraint w/ an unexpected name (e.g. `runs_chk_1`) & the old `status` constraint remains around. Upgrading to 1.6.0 then drops the `status` constraint, leaving behind `runs_chk_1`)
* For a new user who launches the server, we end up doing the same thing - creating a new constraint here with name like `runs_chk_1`, and then dropping the old status constraint in the following migration.

Also, my status column weirdly is still a VARCHAR on a fresh MySQL DB, maybe this is expected?

```
  `status` varchar(9) DEFAULT NULL,
```

Reply:
@AveshCSingh investigated, looks like `constraint_name` isn't a valid argument to the [constructor of Enum](https://docs.sqlalchemy.org/en/13/core/type_basics.html#sqlalchemy.types.Enum.__init__) columns in alembic, which is likely why it's not honored. We think this is ok though - if we need to update RunStatus in the future, we probably don't need to directly reference the constraint name/drop it, but instead can just alter the column with a new set of allowed enum values.

Reply:
I guess the main actionable thing here then is just to remove the `constraint_name` arg as it's confusing.

Reply:
@smurching @AveshCSingh Thanks for catching the `constraint_name` issue. I've removed this no-op parameter from the `Enum` constructor!